### PR TITLE
bookkeeper 4.4.0 (new formula)

### PR DESCRIPTION
**Pull request by hanxue**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/bookkeeper.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    inreplace libexec/"bookkeeper" do |s|
+      s.gsub! /^BINDIR=.*$/, "BINDIR=#{bin}"
+      s.gsub! /^BK_HOME=.*$/, "BK_HOME=#{prefix}"
+      s.gsub! "$BK_HOME/lib", "$BK_HOME"
+    end
+
+    # bin.mkpath
+    # (etc/"zookeeper").mkpath
+    # (var/"log/zookeeper").mkpath
+    # (var/"run/zookeeper/data").mkpath
```
Comment (MikeMcQuaid) [inline Formula/bookkeeper.rb]: If these aren't needed they can be commented out.

Comment (MikeMcQuaid) [inline Formula/bookkeeper.rb]: Er, deleted.

File: Formula/bookkeeper.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  end
+
+  def caveats; <<-EOS.undent
+    You need to initialize zookeper before running bookkeeper. Use the command:
```
Comment (MikeMcQuaid) [inline Formula/bookkeeper.rb]: `zookeper` to `zookeeper`.

File: Formula/bookkeeper.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      bookkeeper shell metaformat
```
Comment (MikeMcQuaid) [inline Formula/bookkeeper.rb]: What does this do? Can we run it automatically?

File: Formula/bookkeeper.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    You may also want to set the JAVA_HOME environment to a suitable value.
```
Comment (MikeMcQuaid) [inline Formula/bookkeeper.rb]: Remove this line.

**Comment by MikeMcQuaid** [comment]
> Does your submission pass brew audit --strict --online <formula> (after doing brew install <formula>)?

@hanxue Needs a test for that. Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by DomT4** [comment]
Ping?

**Comment by DomT4** [comment]
Closing as dead but will take a fresh resubmission with the comments addressed. Thank you for the submission to Homebrew nonetheless ❤️.

**Pull request closed without merging**